README.md

```diff
@@ -1,3 +1,31 @@
 # go-library is a monorepo of modules
 
 The go-library is a monorepo of golang modules designed for reusability within CLI and daemon applications.
+
+## Conventions
+
+A module should be self contained within its own folder off the root of the repository and contain its own `go.mod`.
+
+For example:
+```
+mod1/
+    pkg
+    go.mod
+mod2/
+    pkg
+    go.mod
+```
+
+* All modules MUST have a `pkg` directory with the exported interfaces and functions
```

Comment:
Aren't these all implied to be public packages and APIs? what is the pkg directory signaling? It seems to just expand import paths.

Reply:
Even a public library could depend on internal functions that are not designed to be consumed by an external client. I think we need to maintain a distinction between what functions are designed to be public and which are private; which suggests that each module should have a `pkg` and may have an `internal`.

Reply:
I think "MUST" is probably the wrong wording here. If the package is simple, it's reasonable to just have `.go` files directly in the top-level directory. The `pkg` and `internal` conventions can be more of an organizational tool for complex packages.

Reply:
Thats an OK distinction.  if simple; flat `.go` files are OK.  But when it gets more complex and we need to distinguish between public/private; it gets moved as a breaking change.

Reply:
Doesn't Public vs private scope cover the use cases here? If something isn't designed to be consumed outside of the package don't export it beyond the package scope.

Reply:
```suggestion
* All modules MAY define flat `.go` files in the package directory with the exported interfaces and functions. Complex packages MAY also have a `pkg` directory for organizational purposes.
```

Reply:
Also true.  I updated the guidelines accordingly; leaving these as suggestions that the author can determine based the complexity of the module.